machine/v1/types_controlplanemachineset.go

```diff
@@ -345,6 +352,14 @@ type OpenStackFailureDomain struct {
 	RootVolume *RootVolume `json:"rootVolume,omitempty"`
 }
 
+// NutanixFailureDomainReference refers to the failure domain of the Nutanix platform.
+type NutanixFailureDomainReference struct {
+	// name of the failure domain in which the nutanix machine provider will create the VM.
+	// Failure domains are defined in a cluster's config.openshift.io/Infrastructure resource.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
```

Comment:
would the same validation on `name` in https://github.com/openshift/api/pull/1578/files#diff-07b264a49084976b670fb699badaca1795027d6ea732a99226a5388104f6174fR1435-R1439 apply here?

Reply:
The name is used as the reference, the cpmso and mapi-nutanix controller will validate the referenced FD name defined in the Infrastructure resource. So I think we don't need to apply pattern validation here.

Reply:
You do need to apply pattern validation here the same as in the infrastructure resource please.

Where did the name validation come from in the infrastructure?

Reply:
Changed.